File: sdks/java/Http.java
// Copyright 2026 J3nna Technologies, LLC
// SPDX-License-Identifier: Apache-2.0
//
// Minimal JSON-over-HTTP helper for the SDK modules (mirrors the Python stdlib urllib helpers).
// java.net.http.HttpClient + gson, no external deps beyond gson.

import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

final class Http {
    private static final HttpClient CLIENT = HttpClient.newBuilder()
            .connectTimeout(Duration.ofSeconds(10)).build();

    private Http() {}

    static JsonObject get(String url, double timeoutSec) throws IOException, InterruptedException {
        HttpRequest req = HttpRequest.newBuilder(URI.create(url))
                .timeout(Duration.ofMillis((long) (timeoutSec * 1000)))
                .GET().build();
        HttpResponse<String> r = CLIENT.send(req, HttpResponse.BodyHandlers.ofString());
        return JsonParser.parseString(r.body()).getAsJsonObject();
    }

    static JsonObject post(String url, JsonElement body, double timeoutSec)
            throws IOException, InterruptedException {
        HttpRequest req = HttpRequest.newBuilder(URI.create(url))
                .timeout(Duration.ofMillis((long) (timeoutSec * 1000)))
                .header("content-type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString(body.toString()))
                .build();
        HttpResponse<String> r = CLIENT.send(req, HttpResponse.BodyHandlers.ofString());
        return JsonParser.parseString(r.body()).getAsJsonObject();
    }
}

File: sdks/rust/src/http.rs
// Copyright 2026 J3nna Technologies, LLC
// SPDX-License-Identifier: Apache-2.0
//! Minimal blocking JSON HTTP helpers over ureq. All mesh endpoints are plain HTTP/JSON.

use std::time::Duration;

use crate::error::{Error, Result};

fn agent(timeout: f64) -> ureq::Agent {
    ureq::AgentBuilder::new()
        .timeout(Duration::from_secs_f64(timeout))
        .build()
}

pub fn get_json(url: &str, timeout: f64) -> Result<serde_json::Value> {
    let resp = agent(timeout)
        .get(url)
        .call()
        .map_err(|e| Error::Http(format!("GET {url}: {e}")))?;
    resp.into_json::<serde_json::Value>()
        .map_err(|e| Error::Http(format!("GET {url} decode: {e}")))
}

pub fn post_json(url: &str, body: &serde_json::Value, timeout: f64) -> Result<serde_json::Value> {
    let resp = agent(timeout)
        .post(url)
        .set("content-type", "application/json")
        .send_json(body.clone())
        .map_err(|e| Error::Http(format!("POST {url}: {e}")))?;
    resp.into_json::<serde_json::Value>()
        .map_err(|e| Error::Http(format!("POST {url} decode: {e}")))
}
